### ISSUE TITLE
gatsby-rehype-inline-images missing images

### ISSUE THREAD

**Issue by marcoSven** [styxlab/gatsby-theme-try-ghost#202]
- Missing inline images in the build - locally and Netlify
- Images are shown in development but they are not sourced from /static folder
- Lots of build failures in Netlify related to sharp - could be related to memory limit?
- Inline images in the Ghost CMS are uploaded or inserted via Unsplash - not sure if that matters. 

Here is an example 
https://github.com/marcoSven/gatsby-starter-try-ghost-1 -
CMS - https://ghostwriterdev.bettinalimaco.com/organising-content/
 the first image made it for some reason the following two did not - I deployed multiple times cleared cache
Netlify -  https://zealous-wescoff-a5d95e.netlify.app/organising-content/

**Comment by marcoSven** [comment]
I guess missing is not correct looks like they are actually processed and stored in the static folder, there is an issue with the lazy loading. 

> build failures in Netlify related to sharp - could be related to memory limit?
I guess this is a separate issue.

**Comment by styxlab** [comment]
This is strange, and I would need (temporary) access keys to your Ghost CMS to analyze the issue further. What  I found is:

- The missing images are generated and processed (as you found out already)
- The image is present in the html, but the fluid container is resizing them to 0x0, that's why you can't see them (it's not a problem with lazy loading)
- As a first test, I saved the image 'bettina.jpg' to my local machine, embedded it into a page in the headless CMS and rebuild: the image is coming through as expected.

Do you have any custom CSS in your Ghost CMS that could negatively interfere?

**Comment by styxlab** [comment]
I'm sorry to hear you are running into issues. I was able to reproduce the issue internally. Here are my findings:

- Your pictures are processed (as you found out yourself, it's not a problem with sharp, you can ignore those messages).
- The pictures are included in the html and loaded (it's not a problem with lazy loading).
- The fluid container is wrongly resized to 0x0 px, so the picture is present but you can't see a 0x0 image.

Unfortunately, this cannot be fixed easily as the `gatsby-image-fluid` containers are no always working as expected.

A workaround is to ensure that all your pictures have a minimum width of 700px. That way the fluid container is able to do the resizing correctly.

**Comment by marcoSven** [comment]
Thank you for looking into this

> (it's not a problem with lazy loading).

> The fluid container is wrongly resized to 0x0 px, so the picture is present but you can't see a 0x0 image

You are right 

> A workaround is to ensure that all your images have a minimum width of 700px

Not sure if this will work I used larger images than 700px and experienced the same issue. Take a look at the 3rd image in the example /content/images/size/w1600/2020/10/cousins.jpg

**Comment by marcoSven** [comment]
@styxlab if you want to give it a try see info below. Inline images are still mostly not showing or in my last attempt when I did a clean install, content images are not using the local version of the images. Looking at [demo](https://demo.jamify.org/apps-integrations/) only the featured image use the local image in the static directory.
 
cc35e2b7b90b796eafb2bd96f4 https://ghostwriter.bettinalimaco.com

**Comment by styxlab** [comment]
The `gatsby-inline-image` plugin uses fluid elements which need to be wrapped in a container that is not fluid. As the standard ghost css is using an fluid element for the container, browser cannot determine correct width and height. This is not a problem, if the image is placed directly in that container (which is the standard way without the inline images). This problem got undetected on my side as I am using the `gatsby-theme-ghost-toc` plugin which adds some additional styles. So, fix is to add the correct stylings to the `gatsby-inline-image` plugin.

**Comment by marcoSven** [comment]
>  I am using the gatsby-theme-ghost-toc plugin

Yes, I noticed that yesterday I was about to report this here. 

There are 3 different inline images insert options in ghost regular wide and full. I have not tested full but all wide images are not visible because of the css issue you mentioned. 

What caused some confusion here is build issue #204  When I develop or build locally on my mac the images are not sourced from the static directory. There is no issue in the gatsby cloud. 

>  A fix is on its way.

Thank you! Great, we got this resolved.
